### PR TITLE
clear metrics api add

### PR DESCRIPTION
**Pull request by shramroy**

### CODE REVIEW

**Review by ajbalogh — CHANGES_REQUESTED** [review]
This link provides additional information about `choice` https://github.com/open-traffic-generator/openapiart/blob/main/MODELGUIDE.md#keyword-limitations

File: control/control.yaml [hunk 1]
```diff
@@ -51,6 +51,36 @@ components:
           type: string
           enum: [start, stop, pause]
 
+    Clear.Metrics:
+      description: >-
+        Clear metrics.
+      type: object
+      required: [metrics_choice]
```
[the remark this answers is not in the document]
Comment (ankur-sheth) [reply control/control.yaml:58]: Agree.  `metrics_choice` doesn't seem to be clear.  We can keep it as `choice` as Andy suggests.

File: control/control.yaml [hunk 2]
```diff
@@ -51,6 +51,36 @@ components:
...
+      properties:
+        port_names:
+          description: >-
+            The names of port objects to.
```
Comment (ankur-sheth) [inline control/control.yaml:62]: comment is incomplete. Perhaps rephrase as "The names of port objects for which metrics will be cleared"?

File: control/control.yaml [hunk 3]
```diff
@@ -51,6 +51,36 @@ components:
...
+            An empty or null list will clear metrics of all ports.
+          type: array
+          items:
+            type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+        flow_names:
+          description: >-
+            The names of flow objects to.
```
Comment (ankur-sheth) [inline control/control.yaml:71]: Same comment as above.  Incomplete comment.

**Review by ankur-sheth — CHANGES_REQUESTED** [review]

**Comment by ankur-sheth** [comment]
Also a general comment, please provide pseudo-code and link to issue(s) in the PR header.

**Comment by shramroy** [comment]
Way to use clear_metrics api - 
metrics = api.clear_metrics()
metrics.choice = 'ports'
metrics.ports = ["p1", "p2"]
api.clear_metrics(metrics)

File: control/control.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (shramroy) [reply control/control.yaml]: If I put choice then "all" has to be an defined object type as per model definition 
 # demonstrates how to model a choice
  components:
    schemas:
    Choice.Object:
      type: object
      properties:
        choice:
          type: string
          enum: [a, b, c]
          default: a
        a:
          $ref: '#/components/schemas/Choice.A'
        b:
          $ref: '#/components/schemas/Choice.B'
        c:
          $ref: '#/components/schemas/Choice.C'




But with the choice "all" I don't want to give any other input. 
If the "metrics_choice" term is not convincing I changed it to "options". Will it be okay?

File: control/control.yaml (hunk 3 above)
Comment (shramroy) [reply control/control.yaml]: modified

File: control/control.yaml (hunk 2 above)
Comment (shramroy) [reply control/control.yaml]: modified

**Review by ankur-sheth — APPROVED** [review]

**Review by ajbalogh — COMMENTED** [review]
With the use cases detailed in PR (https://github.com/open-traffic-generator/models/pull/234) where is the need for this API?

Specifically what are the use case(s) where a user would want to clear metrics that isn't met by PR 234.

**Comment by ankur-sheth** [comment]
> With the use cases detailed in PR (#234) where is the need for this API?
> 
> Specifically what are the use case(s) where a user would want to clear metrics that isn't met by PR 234.

Agree with @ajbalogh.  Perhaps we can park this till we really have a use-case for it?

**Comment by ajbalogh** [comment]
Closing this PR until specific use cases dictate a need for it.

**Pull request closed without merging**